src/invoice/signature.rs

```diff
@@ -415,9 +419,32 @@ impl SecretKeyFile {
     }
 }
 
+pub enum LabelMatch<'a> {
+    FullMatch(&'a str),
+    PartialMatch(&'a str),
+    None,
+}
+
+impl<'a> Default for LabelMatch<'a> {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
 impl SecretKeyStorage for SecretKeyFile {
-    fn get_first_matching(&self, role: &SignatureRole) -> Option<&SecretKeyEntry> {
-        self.key.iter().find(|k| k.roles.contains(role))
+    fn get_first_matching(
+        &self,
+        role: &SignatureRole,
+        label_match: LabelMatch,
```

Comment:
Would it be better if we did `Option<LabelMatch>` and drop the `LabelMatch::None`? That might be a bit cleaner and makes it so people don't have to import `LabelMatch` if they aren't using it

Reply:
yeah, I thought its good to have a `Default` implementation that can be used in test files etc rather than just using `None`
but let me know if I can change this to `Option<LabelMatch >` !!

Reply:
I changed it to Option<LabelMatch> and removed `LabelMatch::None`